src/ThriveWorkerUnit.sol

```diff
@@ -0,0 +1,149 @@
+ */
+contract ThriveWorkerUnit {
+    using EnumerableSet for EnumerableSet.AddressSet;
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    address public immutable moderator;
+    address public immutable rewardToken;
+    uint256 public immutable rewardAmount;
+    uint256 public immutable maxRewards;
+    uint256 public immutable validationRewardAmount;
+    uint256 public maxCompletionsPerUser;
+    uint256 public deadline;
+
+    IBadgeQuery public badgeQuery;
+    bool public ready;
+
+    string public validationMetadata;
+    string public metadataVersion;
+    string public metadata;
+
+    EnumerableSet.AddressSet private validators;
+    EnumerableSet.Bytes32Set private requiredBadges;
+    mapping(address => uint256) public completions;
+
+    event Initialized();
+    event ConfirmationAdded(
+        address indexed contributor,
+        string validationMetadata,
+        uint256 rewardAmount,
+        address indexed validator
+    );
+
+    event ConfigurationUpdated(string field, uint256 newValue);
+    event MetadataUpdated(string field, string newValue);
+
+    modifier onlyModerator() {
+        require(msg.sender == moderator, "Only the moderator can perform this action");
+        _;
+    }
+
+    modifier onlyValidator() {
+        require(validators.contains(msg.sender), "Only a validator can perform this action");
+        _;
+    }
+
+    modifier onceReady() {
+        require(ready, "Contract is not ready");
+        _;
+    }
+
+    constructor(
+        address _moderator,
+        address _rewardToken,
+        uint256 _rewardAmount,
+        uint256 _maxRewards,
+        uint256 _deadline,
+        uint256 _maxCompletionsPerUser,
+        address[] memory _validators,
+        string memory _validationMetadata,
+        string memory _metadataVersion,
+        string memory _metadata,
+        address _badgeQuery
+    ) {
+        require(_moderator != address(0), "Moderator address is required");
+        require(_badgeQuery != address(0), "BadgeQuery address is required");
+        require(_deadline > block.timestamp, "Deadline must be in the future");
+
+        moderator = _moderator;
+        rewardToken = _rewardToken;
+        rewardAmount = _rewardAmount;
+        maxRewards = _maxRewards;
+        validationRewardAmount = _rewardAmount / 10; // Validators = 10% of contributor reward
+        deadline = _deadline;
+        maxCompletionsPerUser = _maxCompletionsPerUser;
+
+        for (uint256 i = 0; i < _validators.length; i++) {
+            validators.add(_validators[i]);
+        }
+
+        validationMetadata = _validationMetadata;
+        metadataVersion = _metadataVersion;
+        metadata = _metadata;
+        badgeQuery = IBadgeQuery(_badgeQuery);
+    }
+
+    function initialize() external payable onlyModerator {
+        uint256 requiredEther = validators.length() * validationRewardAmount;
```

Comment:
hmm one validator performs the validation, we should ensure that we have enough ether for maxRewards

Reply:
I managed this by letting the validators array to be initialzied once and calculated example:
`validationRewardAmount = 10% of rewardAmount per contributor` and then 
`uint256 requiredEther = validators.length() * validationRewardAmount;
        uint256 totalRequiredEther = maxRewards + requiredEther;`
and required the amount of sent eth to be more than `totalRequiredEther`

Reply:
hmm this math doesn't seem right, first see comments about   validationRewardAmount = _rewardAmount / 10; // Validator Reward = 10% of contributor reward

I think it should be simply this:
uint256 totalRequiredEther = maxRewards * validationRewardAmount

at most one validator can perform the confirmation of a specific contribution so it would match above